cmake/linker/lld/linker_flags.cmake

```diff
@@ -1,11 +1,20 @@
 # Copyright (c) 2022 Google LLC
 # SPDX-License-Identifier: Apache-2.0
 
-# Since lld is a drop in replacement for ld, we can just use ld's flags
-include(${ZEPHYR_BASE}/cmake/linker/ld/${COMPILER}/linker_flags.cmake OPTIONAL)
+# Since lld is a drop in replacement for ld, we can just use ld's flags as a base
+# and adjust for lld specifics afterwards.
+include(${ZEPHYR_BASE}/cmake/linker/ld/linker_flags.cmake OPTIONAL)
 
-check_set_linker_property(TARGET linker PROPERTY memusage "${LINKERFLAGPREFIX},--print-memory-usage")
+if(NOT CONFIG_MINIMAL_LIBCPP AND NOT CONFIG_NATIVE_LIBRARY AND NOT CONFIG_EXTERNAL_MODULE_LIBCPP)
+  set_property(TARGET linker PROPERTY cpp_base -lc++ ${LINKERFLAGPREFIX},-z,norelro)
```

Comment:
This seems to tie lld to use with clang? From what I can see, the old code checked the compiler (`CMAKE_C_COMPILER_ID STREQUAL "Clang"`).

Reply:
no, old code only checked for clang in ld, not in lld:
From ld:
https://github.com/zephyrproject-rtos/zephyr/blob/689d1edee1d57f052b1d4572d67618c0b0e2b8a4/cmake/linker/ld/target_base.cmake#L48-L56

From lld:
https://github.com/zephyrproject-rtos/zephyr/blob/689d1edee1d57f052b1d4572d67618c0b0e2b8a4/cmake/linker/lld/target_base.cmake#L42-L50

I try in this series to not have any functional changes, but I do agree that this is not directly a linker issue, but an issue caused by the output of the compiler and therefore should probably be moved to `lld/clang` flags file, but that would be one of the coming PRs :wink: 

Note, the clang check in ld is ensured, because the flag is set in https://github.com/zephyrproject-rtos/zephyr/blob/bcab9417a49989db9dc09a9b9145c1dcfcfe01f1/cmake/linker/ld/clang/linker_flags.cmake